### PR TITLE
Hello World with gRPC

### PR DESCRIPTION
**Pull request by mocsharp**
This sample application demonstrates the use of a gRPC service inside HSDK. 
This project creates a `grpc_server` executable and a `grpc_client` executable.

Note: Holoscan 2.5 is required for this sample application.

### CODE REVIEW

File: applications/grpc_hello_world/entity_client.cc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ENTITY_CLIENT_CC
+#define ENTITY_CLIENT_CC
```
Comment (tbirdso) [inline applications/grpc_hello_world/entity_client.cc:19]: Nitpick: please rename `.cc` files to `.cpp` for consistency and update include guards

```suggestion
#ifndef ENTITY_CLIENT_CPP
#define ENTITY_CLIENT_CPP
```

File: applications/grpc_hello_world/grpc_client.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  std::string server_address_;
+};
+}  // namespace grpc_hello_world
+}  // namespace holoscan
+
+int main(int argc, char** argv) {
+  auto config_path = std::filesystem::canonical(argv[0]).parent_path();
+  config_path += "/grpc_hello_world.yaml";
+  auto app = holoscan::make_application<holoscan::grpc_hello_world::ClientApp>("localhost:50051");
+  app->config(config_path);
```
Comment (tbirdso) [inline applications/grpc_hello_world/grpc_client.cpp:134]: Thoughts on accepting a custom config path to aid testing? Not needed for this app scope?

**Review by tbirdso — COMMENTED** [review]
Demo looks great, only minor nitpicks. Will wait for 2.5 to merge.

File: cmake/grpc_generate_cpp.cmake [hunk 3]
```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 gRPC authors.
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply cmake/grpc_generate_cpp.cmake:1]: Hi @tbirdso, That is where I originally had it. However, I will be adding another application that depends on it, so I moved it to the cmake directory.

[the remark this answers is not in the document]
Comment (tbirdso) [reply cmake/grpc_generate_cpp.cmake:1]: Gotcha, thanks for clarifying. Given that plan I am OK with adding it as a common resource.

**Comment by tbirdso** [comment]
Hi @mocsharp , could you please update on the status of this PR?

**Comment by mocsharp** [comment]
Closing this PR in favor of the upcoming gRPC ETT app.

**Pull request closed without merging**